src/pat/autofocus/autofocus.js

```diff
@@ -1,34 +1,36 @@
-/**
- * Patterns autofocus - enhanced autofocus form elements
- *
- * Copyright 2012-2013 Simplon B.V. - Wichert Akkerman
- */
 import $ from "jquery";
-import registry from "../../core/registry";
+import Base from "../../core/base";
 
-var autofocus = {
+let scheduled_task = null;
+
+export default Base.extend({
     name: "autofocus",
     trigger: ":input.pat-autofocus,:input[autofocus]",
 
-    init: function init() {
+    init() {
+        if (window.self !== window.top) {
+            // Do not autofocus in iframes.
+            return;
+        }
+
         this.setFocus(this.trigger);
-        $(document).on("patterns-injected", function (e) {
-            autofocus.setFocus($(e.target).find(autofocus.trigger));
-        });
-        $(document).on("pat-update", function (e) {
-            autofocus.setFocus($(e.target).find(autofocus.trigger));
+        $(document).on("patterns-injected pat-update", (e) => {
+            this.setFocus($(e.target).find(this.trigger));
         });
     },
-    setFocus: function (target) {
-        var $all = $(target);
-        var $visible = $all.filter(function () {
-            if ($(this).is(":visible")) return true;
-        });
-        setTimeout(function () {
-            $visible.get(0) && $visible.get(0).focus();
-        }, 10);
-    },
-};
 
-registry.register(autofocus);
-export default autofocus;
+    setFocus(target) {
+        // Clear previously scheduled tasks.
+        // setFocus normally operates on the whole DOM anyways and if scoped
+        // (via the event handler in init) the last one wins.
+        // This is OK since only one element can have the focus.
+        clearTimeout(scheduled_task);
```

Comment:
@pilz this is an optimization to call autofocus only once and not multiple times for each pat-autofocus element which can be found in the DOM.